icu4c/source/test/intltest/loctest.cpp

```diff
@@ -2748,4 +2752,68 @@ void LocaleTest::TestBug13554() {
     }
 }
 
+void LocaleTest::TestMoveAssign() {
+    // ULOC_FULLNAME_CAPACITY == 157 (uloc.h)
+    Locale l1("de@collation=phonebook;x="
+              "aaaaabbbbbcccccdddddeeeeefffffggggghhhhh"
+              "aaaaabbbbbcccccdddddeeeeefffffggggghhhhh"
+              "aaaaabbbbbcccccdddddeeeeefffffggggghhhhh"
+              "aaaaabbbbbzz");
+
+    Locale l2;
+    {
+        Locale l3(l1);
+        assertTrue("l1 == l3", l1 == l3);
+        assertTrue("!l3.isBogus()", !l3.isBogus());
+        l2 = std::move(l3);
+        assertTrue("l1 == l2", l1 == l2);
+        assertTrue("l3.isBogus()", l3.isBogus());
```

Comment:
Some compilers might complain that you're calling methods on a move'd instance.  It shouldn't be worse than a warning, though.

Reply:
If that ends up being a problem, we could then remove the tests on the move'd object, but until then I think it's a good idea to try to verify that it really has been modified as we want it to be modified.